app/src/org/commcare/activities/FormAndDataSyncer.java

```diff
@@ -91,6 +93,12 @@ protected void deliverResult(SyncCapableCommCareActivity receiver, FormUploadRes
                             case TRANSPORT_FAILURE:
                                 receiver.handleFormSendResult(Localization.get("sync.fail.bad.network"), false);
                                 break;
+                            case BAD_CERTIFICATE:
+                                CommCareApplication.notificationManager().reportNotificationMessage(
```

Comment:
That makes sense. The issue I was having was that the call back to the receiver already reduced the FormUploadResult to a string so the receiver couldn't determine when to show the notification. So to allow that I created a new handleFormUploadResult method on the receiver that takes the FormUploadResult instead of a string message.

At this point it seemed to make sense to make FormUploadResult a MessageTag and use that to map the enum values to the translatable strings. So I did that and used the mappings from the switch structure to populate the enum mappings.

But then it seemed to make sense to move the entire switch from FormAndDataSyncer to the new method, since most of the cases were simply translating enum to message anyway. That leaves only a little success-based logic in FormAndDataSyncer and otherwise puts the logic all in SyncCapableCommCareActivity.

Finally, I added two more cases to FormUploadRestul for UNSENT and CANCELLED. That allowed me to replace the other two calls in FormAndDataSyncer to handleFormSendResult, such that only handleFormUploadResult is called now.